lib/bin/lwm2m_carrier/include/lwm2m_os.h

```diff
@@ -21,16 +22,79 @@
 #define LWM2M_OS_MAX_TIMER_COUNT 6
 
 #define LWM2M_LOG_LEVEL_NONE 0
-#define LWM2M_LOG_LEVEL_ERR  1
-#define LWM2M_LOG_LEVEL_WRN  2
-#define LWM2M_LOG_LEVEL_INF  3
-#define LWM2M_LOG_LEVEL_TRC  4
+#define LWM2M_LOG_LEVEL_ERR 1
+#define LWM2M_LOG_LEVEL_WRN 2
+#define LWM2M_LOG_LEVEL_INF 3
+#define LWM2M_LOG_LEVEL_TRC 4
+
+/**
+ * @brief Range of the non-volatile storage identifiers used by the library.
+ *
+ * @note  The application MUST NOT use the values within this range for its
+ *        own non-volatile storage management as it could potentially delete
+ *        or overwrite entries used by the library.
+ */
+#define LWM2M_OS_STORAGE_BASE 0xCA00
+#define LWM2M_OS_STORAGE_END 0xCAFF
 
 /**
  * @brief Timer callback function.
  */
 typedef void (*lwm2m_os_timer_handler_t)(void *timer);
 
+/**
+ * @brief List of AT parameters that compose an AT command or response.
+ */
+struct lwm2m_os_at_param_list {
+	size_t param_count;
+	void *params;
+};
+
+/**
+ * @brief AT Command handler.
+ */
+typedef void (*lwm2m_os_at_cmd_handler_t)(void *ctx, char *response);
+
+/**
+ * @brief Download client event IDs.
+ */
+enum lwm2m_os_download_evt_id {
```

Comment:
I'll just link my comment from previous PR here, for consideration: https://github.com/NordicPlayground/fw-nrfconnect-nrf/pull/1313#discussion_r337046714